python/foglamp/services/core/api/configuration.py

```diff
@@ -178,6 +181,75 @@
     return web.json_response(result)
 
 
+async def add_configuration_item(request):
+    """
+    Args:
+        request: A JSON object that defines the config item and has key-pair
+                 (default, type, description, value[optional])
+
+    Returns:
+        Json response with message key
+
+    :Example:
+        curl -d '{"default": "true", "description": "Test description", "type": "boolean"}' -X POST https://localhost:1995/foglamp/category/{new_config_item} --insecure
+        curl -d '{"default": "true", "description": "Test description", "type": "boolean", "value": "false"}' -X POST https://localhost:1995/foglamp/category/test_cat/{new_config_item} --insecure
+    """
+    category_name = request.match_info.get('category_name', None)
+    new_config_item = request.match_info.get('add_config_item', None)
+
+    try:
+        storage_client = connect.get_storage()
+        cf_mgr = ConfigurationManager(storage_client)
+
+        data = await request.json()
+        if not isinstance(data, dict):
+            raise ValueError('Data payload must be a dictionary')
+
+        # if value key is in data then go ahead with data payload and validate
+        # else update the data payload with value key and set its value to default value and validate
+        val = data.get('value', None)
+        if val is None:
+            temp_dict = data
+            temp_dict.update({'value': temp_dict.get('default')})
+            config_item_dict = {new_config_item: temp_dict}
+        else:
+            config_item_dict = {new_config_item: data}
+
+        # validate configuration category value
+        await cf_mgr._validate_category_val(category_val=config_item_dict, set_value_val_from_default_val=False)
+
+        # validate category
+        category = await cf_mgr.get_category_all_items(category_name)
+        if category is None:
+            raise NameError("No such Category found for {}".format(category_name))
```

Comment:
check this before `_validate_category_val`

Reply:
Nope, category check DB call after the validate_config_item keys

Reply:
gotcha! you want to avoid storage call, if supplied config item data is not valid.